src/lsdb/core/search/pixel_search.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from typing import List
+
+import pandas as pd
+from hipscat.catalog.catalog_info import CatalogInfo
+from hipscat.pixel_math import HealpixPixel
+
+from lsdb.core.search.abstract_search import AbstractSearch
+
+
+class PixelSearch(AbstractSearch):
+    """Filter the catalog by HEALPix pixels.
+
+    Filters partitions in the catalog to those that are in a specified pixel set.
+    Does not filter points inside those partitions.
+    """
+
+    def __init__(self, pixels):
+        self.pixels = set(pixels)
+
+    def search_partitions(self, pixels: List[HealpixPixel]) -> List[HealpixPixel]:
+        return [p for p in pixels if p in self.pixels]
```

Comment:
Could we do pixel overlap instead of exact match?

For the use case of this PR, you want exact match, but I would love something like:

`read_hipscat("path", filter=PixelSearch([(3, 34)])`

to limit the catalog to a big chunk of the sky (that's order 3, so would likely contain lots of partitions inside it).